src/main/java/io/cdap/plugin/gcp/dataplex/common/config/DataplexBaseConfig.java

```diff
@@ -217,9 +216,10 @@ public String getServiceAccountFilePath() {
    */
   public String getServiceAccountEmail() throws IOException {
     if (connection.isServiceAccountJson() || connection.getServiceAccountFilePath() != null) {
-      ServiceAccountCredentials credentials =
-        GCPUtils.loadServiceAccountCredentials(getServiceAccount(), isServiceAccountFilePath());
-      return credentials.getClientEmail();
+      try (InputStream inputStream = GCPUtils.openServiceAccount(getServiceAccount(), isServiceAccountFilePath())) {
+        ServiceAccountCredentials credentials = ServiceAccountCredentials.fromStream(inputStream);
```

Comment:
this will fail with external service accounts, but wanted to limit scope of this PR as fixing this likely requires adding a new property to the dataplex config

Reply:
I think this should be fine because external account is specifically used for identity federation and requires a projected token (typically kubernetes, but can be any identity provider) to be on the filesystem. The typical user would not be using an external account file.